node/pkg/p2p/cutover.go

```diff
@@ -0,0 +1,109 @@
+package p2p
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// The format of this time is very picky. Please use the exact format specified by cutOverFmtStr!
+const cutOverTimeStr = "2024-12-31T23:59:59-0000"
+const cutOverFmtStr = "2006-01-02T15:04:05-0700"
+
+// checkForCutOver checks to see if / when we need to cut over to the new quic-v1 based on a preset time. If cutOverTimeStr is set, then this feature is enabled.
+// If the current time is after the configured time, we should start up using the new quic-v1 bootstrap peers immediately. If the configured time is in the future,
+// we need to start up with the existing bootstrap string and cut over to the new quic-v1 at the specified time.
+func checkForCutOver(logger *zap.Logger, bootstrapPeers string, ccqBootstrapPeers string, components *Components) (newBootstrapPeers string, newCcqBootstrapPeers string, err error) {
+	newBootstrapPeers, newCcqBootstrapPeers, delay, err := checkForCutOverImpl(logger, bootstrapPeers, ccqBootstrapPeers, components, cutOverTimeStr, time.Now())
+
+	if delay != time.Duration(0) {
+		// Wait for the cut over time and then panic so we restart with the new quic-v1. TODO: Can we just restart p2p??
+		go func() {
+			time.Sleep(delay)
+			logger.Info("time to cut over to new quic-v1", zap.String("cutOverTime", cutOverTimeStr), zap.String("component", "p2pco"))
+			panic("p2pco: time to cut over to new quic-v1")
```

Comment:
My concerns with executing this via governance are

1. Not every peer is guaranteed to see the signed VAA
2. Anyone offline would not know it happened without storing the governance VAA somewhere
3. Where would you store the governance VAA where we could guarantee access to all clients?
4. That storage location would need to be accessible to all clients until all clients have been upgraded to a release dropping the old QUIC support, otherwise clients could suffer a reversion.

I don't feel the added complexity of governance for a simple (and required) protocol switch adds more to the execution and guarantees than it subtracts. This is fundamentally different from a blockchain where guaranteed data synchronization is a part of the protocol (whereas this is not part of the guardian network)

Reply:
@evan-gray works for me! It was more of a thought that I wondered if we couldn't do it for things in the future as well that are more or less "hard forks".

This is a hard change that is "simple", but there will be more in the future. Especially as we start to tease apart the guardian into clean components. Just trying to see where we can reuse stuff.